oak-core/src/main/java/org/apache/jackrabbit/oak/core/MutableTree.java

```diff
@@ -264,11 +264,14 @@ void setParentAndName(@NotNull MutableTree parent, @NotNull String name) {
      * @param newName   new name for this tree
      */
     boolean moveTo(@NotNull MutableTree newParent, @NotNull String newName) {
-        name = checkNotNull(newName);
-        parent = checkNotNull(newParent);
+        checkNotNull(newName);
+        checkNotNull(newParent);
+        MutableTree oldParent = parent;
         boolean success = nodeBuilder.moveTo(newParent.nodeBuilder, newName);
         if (success) {
-            parent.updateChildOrder(false);
+            name = newName;
+            parent = newParent;
```

Comment:
+1 to fixing this one as well - even though it doesn't fix anything, since the MutableTree (source) is thrown away upon returning from `move` - but it was wrong so far, so good to fix this anyway.

Reply:
Right. I forgot to mention this. I'll add a note to the JIRA issue.

Reply:
And I'm going to add a test for this.

Reply:
Done with https://github.com/apache/jackrabbit-oak/pull/1017/commits/f23db0f19ea64d82f7fb771467ba75626f63aaf7